### PR TITLE
Feature 11 archiving

### PR DESCRIPTION
**Pull request by kaeline**
https://github.com/Metatavu/vp-kuljetus-delivery-info-service-api/issues/11
Site archiving listing filtering
Site updates checks (archived Sites cannot be updated unless they are un-archived in in that update)
env. checks for Site removal (no DELETE endpoint for PRODUCTION env)
removed cascade delete logic for all entities 

Also includes additions to Task fields (status field that forbids deletion of DONE tasks)
https://github.com/Metatavu/vp-kuljetus-delivery-info-service-api/issues/10

### CODE REVIEW

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt [hunk 1]
```diff
@@ -159,6 +154,9 @@ class TasksApiImpl : TasksApi, AbstractApi() {
     override fun deleteTask(taskId: UUID): Uni<Response> = CoroutineScope(vertx.dispatcher()).async {
         loggedUserId ?: return@async createUnauthorized(UNAUTHORIZED)
         val foundTask = taskController.findTask(taskId) ?: return@async createNotFound(createNotFoundMessage(TASK, taskId))
+        if (foundTask.status == TaskStatus.DONE) {
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:157]: Should you also not be able to delete `IN_PROGRESS` tasks?

File: src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightUnitssTestIT.kt [hunk 2]
```diff
@@ -1,9 +1,12 @@
...
 
+import fi.metatavu.vp.deliveryinfo.functional.settings.DefaultTestProfile
 import io.quarkus.test.junit.QuarkusIntegrationTest
+import io.quarkus.test.junit.TestProfile
 
 /**
  * Native tests for Freight Units API
  */
 @QuarkusIntegrationTest
+@TestProfile(DefaultTestProfile::class)
```
Comment (villejuutila) [inline src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightUnitssTestIT.kt:11]: Typo in this file name. Change to `NativeFreightUnitsTestIT`

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/sites/SitesApiImpl.kt [hunk 3]
```diff
@@ -69,14 +73,25 @@ class SitesApiImpl: SitesApi, AbstractApi() {
             return@async createBadRequest(INVALID_REQUEST_BODY)
         }
         val existingSite = siteController.findSite(siteId) ?: return@async createNotFound(createNotFoundMessage(SITE, siteId))
+
+        if (existingSite.archivedAt != null && site.archivedAt != null) {
+            return@async createConflict("Cannot update archived site")
```
Comment (villejuutila) [inline src/main/kotlin/fi/metatavu/vp/deliveryinfo/sites/SitesApiImpl.kt:78]: Is it conflict or bad request?

**Review by villejuutila — CHANGES_REQUESTED** [review]

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt (hunk 1 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/vp/deliveryinfo/tasks/TasksApiImpl.kt:157]: No, it is correct, only DONE tasks cannot be removed

File: src/main/kotlin/fi/metatavu/vp/deliveryinfo/sites/SitesApiImpl.kt (hunk 3 above)
Comment (kaeline) [reply src/main/kotlin/fi/metatavu/vp/deliveryinfo/sites/SitesApiImpl.kt:78]: I would personally consider it a conflict. The request is valid, but the state of the target resource (it being archived) is what interferes and makes it un-updatable.
For instance, the quote from mozilla docs:
```
The HTTP 409 Conflict response status code indicates a request conflict with the current state of the target resource.
Conflicts are most likely to occur in response to a [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) request. For example, you may get a 409 response when uploading a file that is older than the existing one on the server, resulting in a version control conflict.
```
Don't you think it fits?

Comment (kaeline) [reply src/main/kotlin/fi/metatavu/vp/deliveryinfo/sites/SitesApiImpl.kt:78]: Plus the other project with archiving already uses the same status codes I think so we either accept this or change that project status codes as well

File: src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightUnitssTestIT.kt (hunk 2 above)
Comment (kaeline) [reply src/native-test/kotlin/fi/metatavu/vp/deliveryinfo/functional/NativeFreightUnitssTestIT.kt]: fixed the typo

**Review by villejuutila — APPROVED** [review]

**Pull request merged**